rxandroidble/src/main/java/com/polidea/rxandroidble/internal/connection/RxBleConnectionImpl.java

```diff
@@ -195,6 +204,10 @@ public int getMtu() {
     @Override
     public Observable<Observable<byte[]>> setupIndication(@NonNull BluetoothGattCharacteristic characteristic,
                                                           @NonNull NotificationSetupMode setupMode) {
+        illegalOperationChecker.checkAnyPropertyMatches(
```

Comment:
I am actually thinking wether or not it would be better to have something like:
```
if (illegalOperationChecker.checkAnyPropertyMatches()) {
  return ... // normal stuff
} else {
  return Observable.error(new IllegalOperationException(/* construct */));
}
```
It would be then consistent no matter how the user would call the function `RxBleConnection.setupIndication(UUID)` / `RxBleConnection.setupIndication(BluetoothGattCharacteristic)`. Because now calling `RxBleConnection.setupIndication(BluetoothGattCharacteristic)` will just throw.

It would need a different abstraction though. @uKL any thoughts?